File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0; int candidate = 0;
        
        for (int num : nums) {
            if (count == 0) candidate = num;    
            count = (num == candidate) ? count + 1 : count - 1;
        }
        return candidate;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        map = {}
        for n in nums:
            if n not in map:
                map[n] = 0
            map[n] += 1
            print(map)
        
        return max(map, key=map.get)
        